tests/components/fully_kiosk/test_diagnostics.py

```diff
@@ -0,0 +1,41 @@
+"""Test the Fully Kiosk Browser diagnostics."""
+from unittest.mock import MagicMock
+
+from aiohttp import ClientSession
+
+from homeassistant.components.diagnostics.const import REDACTED
+from homeassistant.components.fully_kiosk.const import DOMAIN
+from homeassistant.components.fully_kiosk.diagnostics import (
+    DEVICE_INFO_TO_REDACT,
+    SETTINGS_TO_REDACT,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_device
+
+
+async def test_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSession,
+    mock_fully_kiosk: MagicMock,
+    init_integration: MockConfigEntry,
+) -> None:
+    """Test Fully Kiosk diagnostics."""
+
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get_device({(DOMAIN, "abcdef-123456")})
+
+    diagnostics = await get_diagnostics_for_device(
+        hass, hass_client, init_integration, device
+    )
+
+    assert diagnostics
+    for key in DEVICE_INFO_TO_REDACT:
+        assert diagnostics[key] == REDACTED
+    for key in SETTINGS_TO_REDACT:
+        assert (
+            diagnostics["settings"][key] == REDACTED
+            or diagnostics["settings"][key] == ""
+        )
```

Comment:
We generally would like to see the full diagnostic output here, so we can see if there is anything more we want to redact.

Reply:
Here's a dump from my test system:
```
{
  "home_assistant": {
    "installation_type": "Home Assistant Core",
    "version": "2022.9.0.dev0",
    "dev": true,
    "hassio": false,
    "virtualenv": true,
    "python_version": "3.9.5",
    "docker": false,
    "arch": "x86_64",
    "timezone": "America/Indiana/Indianapolis",
    "os_name": "Linux",
    "os_version": "5.11.0-49-generic",
    "run_as_root": false
  },
  "custom_components": {},
  "integration_manifest": {
    "domain": "fully_kiosk",
    "name": "Fully Kiosk Browser",
    "config_flow": true,
    "documentation": "https://www.home-assistant.io/integrations/fullykiosk",
    "requirements": [
      "python-fullykiosk==0.0.11"
    ],
    "dependencies": [],
    "codeowners": [
      "@cgarwood"
    ],
    "iot_class": "local_polling",
    "dhcp": [
      {
        "registered_devices": true
      }
    ],
    "is_built_in": true
  },
  "data": {
    "deviceName": "Amazon Fire",
    "batteryLevel": 100,
    "isPlugged": true,
    "SSID": "\"freewifi\"",
    "Mac": "**REDACTED**",
    "ip4": "**REDACTED**",
    "ip6": "**REDACTED**",
    "hostname4": "**REDACTED**",
    "hostname6": "**REDACTED**",
    "wifiSignalLevel": 8,
    "isMobileDataEnabled": true,
    "screenOrientation": 90,
    "screenBrightness": 0,
    "screenLocked": false,
    "screenOn": true,
    "batteryTemperature": 28,
    "plugged": true,
    "keyguardLocked": false,
    "locale": "en_US",
    "serial": "**REDACTED**",
    "build": "cm_douglas-userdebug 5.1.1 LMY49M 731a881f9d test-keys",
    "androidVersion": "5.1.1",
    "webviewUA": "Mozilla/5.0 (Linux; Android 5.1.1; KFDOWI Build/LMY49M; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/81.0.4044.117 Safari/537.36",
    "motionDetectorStatus": 0,
    "isDeviceAdmin": true,
    "isDeviceOwner": false,
    "internalStorageFreeSpace": 11675373568,
    "internalStorageTotalSpace": 12938534912,
    "ramUsedMemory": 993021952,
    "ramFreeMemory": 447107072,
    "ramTotalMemory": 1440129024,
    "appUsedMemory": 21080480,
    "appFreeMemory": 62805552,
    "appTotalMemory": 83886080,
    "displayHeightPixels": 800,
    "displayWidthPixels": 1280,
    "isMenuOpen": false,
    "topFragmentTag": "",
    "isInDaydream": false,
    "isRooted": true,
    "isLicensed": true,
    "isInScreensaver": true,
    "kioskLocked": true,
    "isInForcedSleep": false,
    "maintenanceMode": false,
    "kioskMode": true,
    "startUrl": "**REDACTED**",
    "currentTabIndex": 0,
    "mqttConnected": true,
    "deviceID": "**REDACTED**",
    "appVersionCode": 875,
    "appVersionName": "1.42.5",
    "androidSdk": 22,
    "deviceModel": "KFDOWI",
    "deviceManufacturer": "amzn",
    "foregroundApp": "de.ozerov.fully",
    "currentPage": "**REDACTED**",
    "lastAppStart": "8/24/2022 1:00:46 AM",
    "sensorInfo": [
      {
        "type": 8,
        "name": "PROXIMITY",
        "vendor": "MTK",
        "version": 1,
        "accuracy": -1
      },
      {
        "type": 5,
        "name": "LIGHT",
        "vendor": "MTK",
        "version": 1,
        "accuracy": 3,
        "values": [
          41,
          0,
          0
        ],
        "lastValuesTime": 1661353412529,
        "lastAccuracyTime": 1661317247048
      }
    ],
    "settings": {
      "mdmSystemUpdatePolicy": "0",
      "tapsToPinDialogInSingleAppMode": "7",
      "movementBeaconList": "",
      "showMenuHint": true,
      "authUsername": "",
      "motionSensitivity": "90",
      "remoteAdminPassword": "**REDACTED**",
      "launchOnBoot": true,
      "kioskAppBlacklist": "",
      "preventSleepWhileScreenOff": false,
      "folderCleanupTime": "",
      "touchInteraction": true,
      "showBackButton": true,
      "volumeLimits": "",
      "addressBarBgColor": -2236963,
      "mdmSystemAppsToEnable": "",
      "rebootTime": "",
      "appLauncherTextColor": -16777216,
      "kioskWifiPin": "",
      "wifiSSID": "",
      "keepSleepingIfUnplugged": false,
      "microphoneAccess": false,
      "barcodeScanBroadcastExtra": "",
      "cacheMode": "-1",
      "screensaverBrightness": "0",
      "knoxDisableUsbDebugging": false,
      "disableOutgoingCalls": false,
      "desktopMode": false,
      "thirdPartyCookies": true,
      "knoxDisableBluetoothTethering": false,
      "protectedContent": false,
      "safeBrowsing": false,
      "screensaverOtherAppIntent": "",
      "mqttBrokerPassword": "**REDACTED**",
      "disablePowerButton": true,
      "screensaverFullscreen": true,
      "inUseWhileKeyboardVisible": false,
      "actionBarTitle": "Fully Kiosk Browser",
      "urlBlacklist": "",
      "pageTransitions": false,
      "enableQrScan": false,
      "playAlarmSoundUntilPin": false,
      "appBlockReturnIntent": "",
      "knoxDisableBackup": false,
      "showAppLauncherOnStart": false,
      "appLauncherScaling": "100",
      "appToRunInForegroundOnStart": "",
      "usageStatistics": true,
      "restartOnCrash": true,
      "tabsFgColor": -16777216,
      "knoxDisableGoogleAccountsAutoSync": false,
      "appToRunOnStart": "",
      "sleepOnPowerDisconnect": false,
      "knoxDisableDeveloperMode": false,
      "remotePdfFileMode": "0",
      "keepOnWhileFullscreen": true,
      "mainWebAutomation": "",
      "deleteHistoryOnReload": false,
      "mdmMinimumPasswordLength": "5",
      "autoplayVideos": false,
      "mdmApkToInstall": "",
      "showQrScanButton": false,
      "startURL": "**REDACTED**",
      "loadOverview": false,
      "showNewTabButton": false,
      "screensaverOtherApp": false,
      "advancedKioskProtection": false,
      "mdmDisableKeyguard": false,
      "searchProviderUrl": "https://www.google.com/search?q=",
      "defaultWebviewBackgroundColor": -1,
      "motionDetectionAcoustic": false,
      "reloadOnScreenOn": false,
      "knoxDisableMultiUser": false,
      "enableBackButton": true,
      "mdmDisableUsbStorage": false,
      "actionBarBgUrl": "",
      "setWifiWakelock": false,
      "lockSafeMode": false,
      "knoxHideStatusBar": false,
      "ignoreMotionWhenScreensaverOnOff": false,
      "showTabCloseButtons": true,
      "kioskWifiPinCustomIntent": "",
      "knoxDisablePowerSavingMode": false,
      "killOtherApps": false,
      "deleteWebstorageOnReload": false,
      "knoxDisableMtp": false,
      "knoxDisablePowerOff": false,
      "mdmAppsToDisable": "",
      "barcodeScanIntent": "",
      "knoxDisableGoogleCrashReport": false,
      "actionBarFgColor": -1,
      "knoxDisablePowerButton": false,
      "barcodeScanTargetUrl": "",
      "audioRecordUploads": false,
      "fadeInOutDuration": "200",
      "autoplayAudio": false,
      "showPrintButton": false,
      "remoteAdminCamshot": true,
      "knoxDisableTaskManager": false,
      "kioskExitGesture": "2",
      "kioskAppWhitelist": "de.badaix.snapcast",
      "remoteAdminScreenshot": true,
      "enableVersionInfo": true,
      "removeStatusBar": false,
      "recreateTabsOnReload": false,
      "showShareButton": false,
      "knoxDisableCellularData": false,
      "injectJsCode": "",
      "screensaverWallpaperURL": "fully://color#000000",
      "removeNavigationBar": false,
      "forceScreenOrientation": "0",
      "showStatusBar": false,
      "inUseWhileAnotherAppInForeground": false,
      "knoxDisableWiFi": false,
      "appLauncherBackgroundColor": -1,
      "mdmAppLockTaskWhitelist": "",
      "webcamAccess": false,
      "loadContentZipFileUrl": "",
      "playAlarmSoundOnMovement": false,
      "knoxDisableClipboard": false,
      "forceImmersive": false,
      "forceShowKeyboard": false,
      "keepScreenOn": true,
      "showCamPreview": false,
      "pauseMotionInBackground": false,
      "timeToScreenOffV2": "0",
      "softKeyboard": true,
      "statusBarColor": 0,
      "mqttBrokerUsername": "**REDACTED**",
      "fileUploads": false,
      "rootEnable": true,
      "screensaverPlaylist": "",
      "knoxDisableUsbTethering": false,
      "launcherApps": "[\n  {\n    \"label\": \"Plex\",\n    \"component\": \"com.plexapp.android\\/com.plexapp.plex.activities.SplashActivity\"\n  },\n  {\n    \"label\": \"Spotify\",\n    \"component\": \"com.spotify.music\\/com.spotify.music.MainActivity\"\n  }\n]",
      "showNavigationBar": false,
      "motionDetection": false,
      "autoImportSettings": true,
      "cloudService": false,
      "webviewDarkMode": "1",
      "killAppsBeforeStartingList": "",
      "websiteIntegration": true,
      "deleteCookiesOnReload": false,
      "clientCaPassword": "",
      "errorURL": "",
      "knoxDisableFactoryReset": false,
      "knoxEnabled": false,
      "knoxDisableCamera": false,
      "playMedia": true,
      "redirectBlocked": false,
      "stopScreensaverOnMotion": true,
      "showTabs": false,
      "mdmDisableADB": true,
      "restartAfterUpdate": true,
      "mqttEventTopic": "$appId/event/$event/$deviceId",
      "fontSize": "100",
      "clearCacheEach": false,
      "mdmDisableVolumeButtons": false,
      "knoxDisableVpn": false,
      "confirmExit": true,
      "ignoreSSLerrors": true,
      "mqttClientId": "",
      "setRemoveSystemUI": false,
      "wifiKey": "",
      "screenOffInDarkness": false,
      "useWideViewport": true,
      "inactiveTabsBgColor": -4144960,
      "knoxDisableWifiDirect": false,
      "geoLocationAccess": false,
      "forceSwipeUnlock": false,
      "graphicsAccelerationMode": "2",
      "authPassword": "",
      "barcodeScanBroadcastAction": "",
      "loadCurrentPageOnReload": false,
      "tabsBgColor": -2236963,
      "enablePullToRefresh": false,
      "skipReloadIfStartUrlShowing": false,
      "motionSensitivityAcoustic": "90",
      "wifiMode": "0",
      "ignoreMotionWhenMoving": false,
      "actionBarCustomButtonUrl": "",
      "knoxDisableAirViewMode": false,
      "kioskHomeStartURL": true,
      "environmentSensorsEnabled": true,
      "disableCamera": false,
      "mdmRuntimePermissionPolicy": "0",
      "knoxDisableRecentTaskButton": false,
      "knoxDisableAirCommandMode": false,
      "remoteAdminSingleAppExit": false,
      "darknessLevel": "10",
      "mqttDeviceInfoTopic": "$appId/deviceInfo/$deviceId",
      "actionBarInSettings": false,
      "remoteAdminLan": true,
      "knoxDisableSDCardWrite": false,
      "movementBeaconDistance": "5",
      "forceSleepIfUnplugged": false,
      "swipeTabs": false,
      "alarmSoundFileUrl": "",
      "knoxDisableStatusBar": false,
      "reloadOnInternet": false,
      "knoxDisableNonMarketApps": false,
      "knoxDisableSettingsChanges": false,
      "barcodeScanListenKeys": false,
      "mdmLockTask": false,
      "timeToScreensaverV2": "899",
      "movementDetection": true,
      "clientCaUrl": "",
      "sleepOnPowerConnect": false,
      "resumeVideoAudio": true,
      "readNfcTag": false,
      "addXffHeader": false,
      "swipeNavigation": false,
      "knoxDisableVolumeButtons": false,
      "screenOnOnMovement": true,
      "timeToClearSingleAppData": "0",
      "knoxHideNavigationBar": false,
      "timeToShutdownOnPowerDisconnect": "0",
      "timeToGoBackground": "0",
      "webviewDragging": true,
      "timeToRegainFocus": "600",
      "knoxDisableUsbHostStorage": false,
      "pauseWebviewOnPause": false,
      "folderCleanupList": "",
      "movementWhenUnplugged": false,
      "actionBarBgColor": -15906911,
      "enableZoom": false,
      "resetZoomEach": false,
      "knoxDisableAndroidBeam": false,
      "reloadOnWifiOn": true,
      "compassSensitivity": "50",
      "touchesOtherAppsBreakIdle": true,
      "reloadPageFailure": "300",
      "actionBarIconUrl": "",
      "knoxDisableBackButton": false,
      "initialScale": "0",
      "remoteAdminFileManagement": true,
      "knoxDisableMicrophoneState": false,
      "knoxDisableFirmwareRecovery": false,
      "errorUrlOnDisconnection": "0",
      "addRefererHeader": true,
      "showHomeButton": true,
      "reloadEachSeconds": "0",
      "accelerometerSensitivity": "80",
      "knoxDisableScreenCapture": false,
      "knoxDisableWifiTethering": false,
      "knoxDisableEdgeScreen": false,
      "forceScreenOrientationGlobal": false,
      "mqttBrokerUrl": "**REDACTED**",
      "forceScreenUnlock": true,
      "knoxDisableClipboardShare": false,
      "wakeupOnPowerConnect": false,
      "waitInternetOnReload": true,
      "barcodeScanInsertInputField": false,
      "knoxDisableAirplaneMode": false,
      "forceOpenByAppUrl": "",
      "disableVolumeButtons": true,
      "disableStatusBar": true,
      "detectIBeacons": false,
      "screensaverDaydream": false,
      "kioskPin": "**REDACTED**",
      "kioskWifiPinAction": "0",
      "videoCaptureUploads": false,
      "stopScreensaverOnMovement": true,
      "motionCameraId": "",
      "disableIncomingCalls": false,
      "knoxDisableSafeMode": false,
      "webHostFilter": false,
      "urlWhitelist": "",
      "progressBarColor": -13611010,
      "disableOtherApps": true,
      "enableTapSound": true,
      "showActionBar": false,
      "volumeLicenseKey": "",
      "ignoreJustOnceLauncher": false,
      "remoteAdmin": true,
      "launcherBgUrl": "",
      "launcherInjectCode": "",
      "mdmDisableSafeModeBoot": true,
      "cameraCaptureUploads": false,
      "disableNotifications": false,
      "sleepSchedule": "",
      "setCpuWakelock": false,
      "knoxDisableAudioRecord": false,
      "enablePopups": false,
      "mdmDisableStatusBar": false,
      "knoxDisableOtaUpgrades": false,
      "mqttEnabled": true,
      "lastVersionInfo": "1.39",
      "textSelection": false,
      "jsAlerts": true,
      "knoxActiveByKiosk": false,
      "webviewDebugging": true,
      "disableHomeButton": true,
      "mdmDisableAppsFromUnknownSources": true,
      "showAddressBar": false,
      "knoxDisableBluetooth": false,
      "localPdfFileMode": "0",
      "playerCacheImages": true,
      "batteryWarning": "20",
      "resendFormData": false,
      "kioskMode": true,
      "mdmDisableScreenCapture": false,
      "singleAppMode": false,
      "enableUrlOtherApps": true,
      "mdmPasswordQuality": "0",
      "navigationBarColor": 0,
      "knoxDisableHomeButton": false,
      "isRunning": true,
      "webviewScrolling": true,
      "motionFps": "5",
      "enableLocalhost": false,
      "reloadOnScreensaverStop": false,
      "customUserAgent": "",
      "screenBrightness": "255",
      "knoxDisableVideoRecord": false,
      "formAutoComplete": true,
      "showProgressBar": true,
      "showRefreshButton": false,
      "showForwardButton": true,
      "knoxDisableHeadphoneState": false,
      "bluetoothMode": "0",
      "timeToClearLauncherAppData": "0",
      "enableFullscreenVideos": true,
      "remoteFileMode": "0",
      "barcodeScanSubmitInputField": false,
      "knoxDisableMultiWindowMode": false,
      "userAgent": "0",
      "singleAppIntent": "",
      "deleteCacheOnReload": false,
      "runInForeground": true,
      "screenOnOnMotion": true
    }
  }
}
```